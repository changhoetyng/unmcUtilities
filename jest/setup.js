const rn = require('react-native')

jest.mock('Linking', () => {
    return {
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      openURL: jest.fn(),
      canOpenURL: jest.fn(),
      getInitialURL: jest.fn(),
    }
})

import mockAsyncStorage from '@react-native-async-storage/async-storage/jest/async-storage-mock';

jest.mock('@react-native-async-storage/async-storage', () => mockAsyncStorage);

module.exports = rn