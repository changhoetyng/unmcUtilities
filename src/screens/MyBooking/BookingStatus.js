import React, {Component} from 'react';
import {View, Text, SafeAreaView, StyleSheet, Dimensions, Button} from 'react-native';
import {theme} from '../../styles/ThemeColour';
import {connect} from 'react-redux'
import moment from 'moment'
import {api} from '../../api/api'

class BookingStatus extends Component {
  constructor(props) {
    super(props);
    this.state = {
      checkInEnabled: false,
    };
  }

  async componentDidMount() {
    await api
        .get(`/student/checkInEnabled/${this.props.bookingStatus._id}`)
        .then((res) => this.setState({checkInEnabled: res.data.checkInStatus}))
        .catch((err)=> console.log(err.response))
  }


  renderButton() {
    
    if(this.state.checkInEnabled) {
      return(
        <Button 
        title="Check in with QR Code"
        onPress={() => this.props.navigation.navigate("CheckIn")}
      />
      )
    } else if(this.props.bookingStatus.status === "checkedIn"){
      return(
        <Button 
        title="Checked out with QR Code"
        onPress={() => this.props.navigation.navigate("CheckIn")}
      />
      )
    } else {
      return(
        <Button 
        title="Not available for check in"
        disabled= {true}
      />
      )
    }
  }

  render() {
    const screenWidth = Dimensions.get('window').width;
    let bookingStatus = this.props.bookingStatus
      return (
        <SafeAreaView style={{backgroundColor: theme.backgroundPrimary, flex: 1}}>
          <View>
            <Text style={[styles.titleStyle, {fontSize: screenWidth * 0.05}]}>
              Details
            </Text>
            <View style={[styles.cardStyle,{fontSize: screenWidth*0.8}]}>
              <Text>Selected Venue: {bookingStatus.venueName}</Text>
              {
                bookingStatus.type === "sportComplex" && <Text>Selected Court: {bookingStatus.subCategoryName}</Text>
              }
               {
                bookingStatus.type === "room" && <Text>Selected Room: {bookingStatus.subCategoryName}</Text>
              }
              <Text>Time: {moment(bookingStatus.bookingTime, 'HH:mm').format(
                  'h:mm a',
                )}{' '}
                -{' '}
                {moment(bookingStatus.bookingTime, 'HH:mm')
                  .add(1, 'hours')
                  .format('h:mm a')}</Text>
              <Text>Date: {moment(
                    bookingStatus.bookingDate,
                  'DD/MM/YYYY',
                ).format('dddd, D MMMM YYYY')}</Text>
            </View>
          </View>
          <View style={{width: "100%",position: 'absolute',bottom:0}}>
          {this.renderButton()}
          
          </View>
        </SafeAreaView>
      );
  }
}

const mapStateToProps = (state) => {
  return {
    bookingStatus: state.bookingReducer.bookingStatus,
  };
};

export default connect(mapStateToProps) (BookingStatus);

const styles = StyleSheet.create({
  cardStyle: {
    width: '100%',
    backgroundColor: 'white',
    padding: 20,
  },
  titleStyle: {
    fontWeight: 'bold',
    color: '#757575',
    marginLeft: 17,
    marginBottom: 10,
  },
});
