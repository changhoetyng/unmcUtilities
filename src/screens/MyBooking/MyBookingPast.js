import React, {Component} from 'react';
import {View, Text, SafeAreaView, StyleSheet} from 'react-native';
import {theme} from '../../styles/ThemeColour';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {FlatList, TouchableOpacity} from 'react-native-gesture-handler';
import {api} from '../../api/api';
import moment from 'moment'
import FullPageLoader from '../../hooks/FullPageLoader'; 

class MyBookingPast extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
      data: [],
    };
  }

  async componentDidMount() {
    this.setState({loading: true});
    await api
      .get('/student/getPast')
      .then((res) => {
        this.setState({data: res.data.booked});
      })
      .catch((err) => {
        console.log(err);
      });
    this.setState({loading: false});
  }

  bookingRenderer(item) {
    return (
      <View style={styles.cardStyle}>
        <Text style={{fontSize: 18}}>{item.venueName}</Text>
        <View
          style={{
            flexDirection: 'row',
            justifyContent: 'space-between',
            alignItems: 'center',
          }}>
          <Text style={{fontSize: 14, color: theme.greyOne}}>
            {item.subCategoryName}
          </Text>
          <TouchableOpacity
            onPress={() => this.props.navigation.navigate('BookingStatus')}>
            <Icon name="arrow-forward" size={35} color={theme.greyOne} />
          </TouchableOpacity>
        </View>
        <Text style={{fontSize: 14, color: theme.greyOne}}>
          {item.bookingDate},{' '}
          {moment(item.bookingTime , 'HH:mm').format('h:mm a')} -{' '}
          {moment(item.bookingTime, 'HH:mm')
            .add(1, 'hours')
            .format('h:mm a')}
        </Text>
      </View>
    );
  }

  render() {
    return (
      <SafeAreaView style={{backgroundColor: theme.backgroundPrimary, flex: 1}}>
        {this.state.loading && <FullPageLoader />}
        <View
          style={{
            marginTop: 20,
            justifyContent: 'center',
            alignItems: 'center',
            flex: 1,
          }}>
          <FlatList
            data={this.state.data}
            contentContainerStyle={{flexGrow: 1}}
            renderItem={({item}) => this.bookingRenderer(item)}
            keyExtractor={(data) => data._id}
            ItemSeparatorComponent={() => <View style={{height: 20}}></View>}
            style={{alignSelf: 'center', width: '100%'}}
          />
        </View>
      </SafeAreaView>
    );
  }
}

export default MyBookingPast;

const styles = StyleSheet.create({
  cardStyle: {
    height: 100,
    width: '90%',
    alignSelf: 'center',
    backgroundColor: theme.backgroundPrimary,
    borderRadius: 10,
    elevation: 10,
    padding: 10,
  },
});
