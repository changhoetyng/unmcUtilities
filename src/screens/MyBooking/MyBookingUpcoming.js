import React, {Component} from 'react';
import {View, Text, SafeAreaView, StyleSheet} from 'react-native';
import {theme} from '../../styles/ThemeColour';
import Icon from 'react-native-vector-icons/MaterialIcons';
import {FlatList, TouchableOpacity} from 'react-native-gesture-handler';
import {api} from '../../api/api';
import moment from 'moment';
import FullPageLoader from '../../hooks/FullPageLoader';
import {connect} from 'react-redux';
import {setBookingStatus} from '../../store/actions/booking';

class MyBookingUpcoming extends Component {
  constructor(props) {
    super(props);
    this.state = {
      loading: false,
      data: [],
    };
  }

  async componentDidMount() {
    this.setState({loading: true});
    await api
      .get('/student/getBooked')
      .then((res) => {
        this.setState({data: res.data.booked});
      })
      .catch((err) => {
        console.log(err);
      });

    this.focusListener = this.props.navigation.addListener(
      'focus',
      async () => {
        this.setState({loading: true});
        await api
          .get('/student/getBooked')
          .then((res) => {
            this.setState({data: res.data.booked});
          })
          .catch((err) => {
            console.log(err);
          });
        this.setState({loading: false});
        //Put your Data loading function here instead of my this.loadData()
      },
    );
    this.setState({loading: false});
  }

  componentWillUnmount() {
    this.focusListener();
  }

  async navigateBookingStatus(item) {
    await this.props.setBookingStatus(item);
    this.props.navigation.navigate('BookingStatus');
  }

  bookingRenderer(item) {
    return (
      <View style={styles.cardStyle}>
        <Text style={{fontSize: 18}}>{item.venueName}</Text>
        <View
          style={{
            flexDirection: 'row',
            justifyContent: 'space-between',
            alignItems: 'center',
          }}>
          <Text style={{fontSize: 14, color: theme.greyOne}}>
            {item.subCategoryName}
          </Text>
          <TouchableOpacity onPress={() => this.navigateBookingStatus(item)}>
            <Icon name="arrow-forward" size={35} color={theme.greyOne} />
          </TouchableOpacity>
        </View>
        {this.renderStatus(item.status)}
        <Text style={{fontSize: 14, color: theme.greyOne}}>
          {item.bookingDate},{' '}
          {moment(item.bookingTime, 'HH:mm').format('h:mm a')} -{' '}
          {moment(item.bookingTime, 'HH:mm').add(1, 'hours').format('h:mm a')}
        </Text>
      </View>
    );
  }

  renderStatus(status) {
    if (status === 'checkedIn') {
      return (
        <Text style={{fontSize: 14, color: theme.greyOne}}>
          Status: <Text style={{color: 'green'}}>Checked In</Text>
        </Text>
      );
    }
    if (status === 'booked') {
      return (
        <Text style={{fontSize: 14, color: theme.greyOne}}>Status: Booked</Text>
      );
    }
  }

  render() {
    return (
      <SafeAreaView style={{backgroundColor: theme.backgroundPrimary, flex: 1}}>
        {this.state.loading && <FullPageLoader />}
        <View
          style={{
            marginTop: 20,
            justifyContent: 'center',
            alignItems: 'center',
            flex: 1,
          }}>
          <FlatList
            data={this.state.data}
            contentContainerStyle={{flexGrow: 1}}
            renderItem={({item}) => this.bookingRenderer(item)}
            keyExtractor={(data) => data._id}
            ItemSeparatorComponent={() => <View style={{height: 20}}></View>}
            style={{alignSelf: 'center', width: '100%'}}
          />
        </View>
      </SafeAreaView>
    );
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    setBookingStatus: (mode) => dispatch(setBookingStatus(mode)),
  };
};

export default connect(null, mapDispatchToProps)(MyBookingUpcoming);

const styles = StyleSheet.create({
  cardStyle: {
    height: 120,
    width: '90%',
    alignSelf: 'center',
    backgroundColor: theme.backgroundPrimary,
    borderRadius: 10,
    elevation: 10,
    padding: 10,
  },
});
