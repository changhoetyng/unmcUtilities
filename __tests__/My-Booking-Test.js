import React from 'react';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';
import BookingStatus from '../src/screens/MyBooking/BookingStatus';
import MyBookingPast from '../src/screens/MyBooking/MyBookingPast';
import MyBookingUpcoming from '../src/screens/MyBooking/MyBookingUpcoming';
import renderer from 'react-test-renderer';
import mock from 'react-native-permissions/mock';
const middlewares = [];
const mockStore = configureStore(middlewares);

jest.mock('@react-native-community/async-storage', () =>
  require('@react-native-community/async-storage/jest/async-storage-mock'),
);

jest.mock('react-native-permissions', () => {
  return mock;
});

const navigation = {
  goBack: jest.fn(),
  navigate: jest.fn(),
  pop: jest.fn(),
  dispatch: jest.fn(),
};

describe('BookingStatus Screen', () => {
  jest.useFakeTimers();
  const store = mockStore({bookingReducer: {bookingStatus:{}}});
  const bookingStatus = renderer.create(
    <Provider store={store}>
      <BookingStatus navigation={navigation}/>
    </Provider>,
  );

  beforeEach(() => {
    jest.resetModules();
    jest.resetAllMocks();
    jest.useFakeTimers();
  });

  it('Render bookingStatus Page', () => {
    expect(bookingStatus.toJSON()).toMatchSnapshot();
  });
});

describe('MyBookingPast Screen', () => {
  jest.useFakeTimers();

  const store = mockStore({bookingReducer: {bookingStatus:{}}});
  const myBookingPast = renderer.create(
    <Provider store={store}>
      <MyBookingPast navigation={navigation} />
    </Provider>,
  );

  beforeEach(() => {
    jest.resetModules();
    jest.resetAllMocks();
    jest.useFakeTimers();
  });

  it('Render myBookingPast Page', async () => {
    expect(myBookingPast.toJSON()).toMatchSnapshot();
  });
});

describe('MyBookingUpcoming Screen', () => {
  jest.useFakeTimers();
  const store = mockStore({bookingReducer: {bookingStatus:{}}});
  const myBookingUpcoming = renderer.create(
    <Provider store={store}>
      <MyBookingUpcoming navigation={navigation}/>
    </Provider>,
  );

  beforeEach(() => {
    jest.resetModules();
    jest.resetAllMocks();
    jest.useFakeTimers();
  });

  it('Render myBookingUpcoming Page', async () => {
    expect(myBookingUpcoming.toJSON()).toMatchSnapshot();
  });
});
