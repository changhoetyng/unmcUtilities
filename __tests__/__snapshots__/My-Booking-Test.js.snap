// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingStatus Screen Render bookingStatus Page 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#DEE4E8",
      "flex": 1,
    }
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "color": "#757575",
            "fontWeight": "bold",
            "marginBottom": 10,
            "marginLeft": 17,
          },
          Object {
            "fontSize": 37.5,
          },
        ]
      }
    >
      Details
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "white",
            "padding": 20,
            "width": "100%",
          },
          Object {
            "fontSize": 600,
          },
        ]
      }
    >
      <Text>
        Selected Venue: 
      </Text>
      <Text>
        Time: 
        Invalid date
         
        -
         
        Invalid date
      </Text>
      <Text>
        Date: 
        Invalid date
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 0,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <View
      accessibilityRole="button"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
            Object {},
          ]
        }
      >
        <Text
          disabled={true}
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "margin": 8,
                "textAlign": "center",
              },
              Object {
                "color": "#cdcdcd",
              },
            ]
          }
        >
          Not available for check in
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;

exports[`MyBookingPast Screen Render myBookingPast Page 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#DEE4E8",
      "flex": 1,
    }
  }
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={
      Object {
        "zIndex": 1100,
      }
    }
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
          "zIndex": 1000,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "display": "flex",
            "height": 150,
            "justifyContent": "space-around",
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/loading.gif",
            }
          }
          style={
            Object {
              "height": "70%",
              "width": "70%",
            }
          }
        />
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Loading
        </Text>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 20,
      }
    }
  >
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      collapsable={false}
      contentContainerStyle={
        Object {
          "flexGrow": 1,
        }
      }
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "alignSelf": "center",
          "width": "100%",
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`MyBookingUpcoming Screen Render myBookingUpcoming Page 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#DEE4E8",
      "flex": 1,
    }
  }
>
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={
      Object {
        "zIndex": 1100,
      }
    }
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#rgba(0, 0, 0, 0.5)",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "space-around",
          "zIndex": 1000,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 10,
            "display": "flex",
            "height": 150,
            "justifyContent": "space-around",
            "width": 150,
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/assets/loading.gif",
            }
          }
          style={
            Object {
              "height": "70%",
              "width": "70%",
            }
          }
        />
        <Text
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          Loading
        </Text>
      </View>
    </View>
  </Modal>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "marginTop": 20,
      }
    }
  >
    <RCTScrollView
      ItemSeparatorComponent={[Function]}
      collapsable={false}
      contentContainerStyle={
        Object {
          "flexGrow": 1,
        }
      }
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      renderScrollComponent={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "alignSelf": "center",
          "width": "100%",
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
